### PR TITLE
Match even more retail actors

### PR DESCRIPTION
**Pull request by cadmic**
EDIT: never mind, no macros here

~~I created a macro for `DebugDisplay_AddObject` since it's rather common and there are a few cases where `if (OOT_DEBUG && ...)` did not work. Unfortunately there are still a few special cases, and it doesn't work in `EnWonderItem_Update` due to same-line issues. Please suggest a better name for the macro, maybe it should really be called `DEBUG_DISPLAY_ADD_OBJECT_DEBUG` but that sounds a bit redundant.~~

~~It's also possible to go a bit farther with the macros, for example this block of code is used a few times:~~
```
    if (this->outOfRangeTimer != 0) {
        if (this->outOfRangeTimer % 2 == 0) {
            DebugDisplay_AddObject(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
                                   this->actor.world.rot.x, this->actor.world.rot.y, this->actor.world.rot.z, 1.0f,
                                   1.0f, 1.0f, 120, 120, 120, 255, 4, play->state.gfxCtx);
        }
    } else {
        DebugDisplay_AddObject(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
                               this->actor.world.rot.x, this->actor.world.rot.y, this->actor.world.rot.z, 1.0f,
                               1.0f, 1.0f, 255, 0, 255, 255, 4, play->state.gfxCtx);
    }
```

~~I'll leave that as future work though.~~

### CODE REVIEW

**Review by Dragorn421 — COMMENTED** [review]
I'm not convinced by the `DEBUG_DISPLAY_ADD_OBJECT` macro, it's too inconsistent to be real

**Comment by cadmic** [comment]
Alright, I've reverted `DEBUG_DISPLAY_ADD_OBJECT`. I still used `OOT_DEBUG &&` everywhere I could (it didn't seem to work in `EnSkj_OcarinaMinigameShortStumpUpdate`, `EnWonderItem_MultitagFree`, and `EnWonderItem_MultitagOrdered`)

File: src/overlays/actors/ovl_En_Encount1/z_en_encount1.c [hunk 1]
```diff
@@ -51,6 +51,8 @@ void EnEncount1_Init(Actor* thisx, PlayState* play) {
     spawnRange = 120.0f + (40.0f * this->actor.world.rot.z);
     this->spawnRange = spawnRange;
 
+    if (1) {}
+
     PRINTF("\n\n");
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Encount1/z_en_encount1.c:56]: kinda curious if this matches as `if(OOT_DEBUG)` surrounding the printfs
(if it doesn not sure if we'd want that)

File: src/overlays/actors/ovl_En_Skj/z_en_skj.c [hunk 2]
```diff
@@ -445,13 +444,17 @@ void EnSkj_Init(Actor* thisx, PlayState* play2) {
...
-            PRINTF("Player_X : %f\n", player->actor.world.pos.x);
-            PRINTF("Player_Z : %f\n", player->actor.world.pos.z);
-            PRINTF("World_X  : %f\n", this->actor.world.pos.x);
-            PRINTF("World_Z  : %f\n", this->actor.world.pos.z);
-            PRINTF("Center_X : %f\n", this->center.x);
-            PRINTF("Center_Z : %f\n\n", this->center.z);
+#if OOT_DEBUG
+            {
+                Player* player = GET_PLAYER(play);
+                PRINTF("Player_X : %f\n", player->actor.world.pos.x);
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Skj/z_en_skj.c:450]: newline police reporting for duty

File: src/overlays/actors/ovl_En_Skj/z_en_skj.c [hunk 3]
```diff
@@ -1502,15 +1505,14 @@ void EnSkj_OfferNextRound(EnSkj* this, PlayState* play) {
...
     if (Message_GetState(&play->msgCtx) == TEXT_STATE_CHOICE && Message_ShouldAdvance(play)) {
         switch (play->msgCtx.choiceIndex) {
-            case 0: // yes
-                player = GET_PLAYER(play);
+            case 0: { // yes
+                Player* player = GET_PLAYER(play);
                 player->stateFlags3 |= PLAYER_STATE3_5; // makes player take ocarina out right away after closing box
                 this->actionFunc = EnSkj_SetupWaitForOcarina;
                 break;
+            }
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Skj/z_en_skj.c:1515]:
```suggestion
            } break;

```

File: src/overlays/actors/ovl_En_Encount1/z_en_encount1.c (hunk 1 above)
Comment (cadmic) [reply src/overlays/actors/ovl_En_Encount1/z_en_encount1.c:56]: It does (for the record, `if (1)` surrounding the printfs matches too)

File: src/overlays/actors/ovl_En_G_Switch/z_en_g_switch.c [hunk 4]
```diff
@@ -80,6 +80,8 @@ void EnGSwitch_Init(Actor* thisx, PlayState* play) {
     s32 pad;
     EnGSwitch* this = (EnGSwitch*)thisx;
 
+    if (play) {}
+
     this->type = (this->actor.params >> 0xC) & 0xF;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_G_Switch/z_en_g_switch.c:85]: does this match with a play2 ( a fake match for recasting the input parameter)
For a while now we've known that they very likely originally passed GameState* to the 4 main actor function and made a temp to recast them in all cases. But we've yet to figure out if it works in ever single case.
If it does, I would prefer a `play2` for now over this

Comment (cadmic) [reply src/overlays/actors/ovl_En_G_Switch/z_en_g_switch.c:85]: It has regalloc issues unfortunately, there's probably a way to get it to work but I wasn't able to figure it out

Comment (fig02) [reply src/overlays/actors/ovl_En_G_Switch/z_en_g_switch.c:85]: thanks for checking

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.c [hunk 5]
```diff
@@ -185,8 +185,7 @@ void EnDntDemo_Judge(EnDntDemo* this, PlayState* play) {
                 case PLAYER_MASK_ZORA:
                 case PLAYER_MASK_GERUDO:
                     rand9 = Rand_ZeroFloat(8.99f);
-                    maskIdx = Player_GetMask(play);
-                    maskIdx--;
+                    maskIdx = (s16)Player_GetMask(play) - 1;
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.c:188]: this looks fake to me (specifically the cast, I think it's more likely it was a short assignment like in main), can you add a `// fake` comment

**Review by Dragorn421 — APPROVED** [review]
after my latest comment is addresses this is good for me

File: src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.c (hunk 5 above)
Comment (cadmic) [reply src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.c:188]: How would you feel about this?
```
// clang-format off
maskIdx = Player_GetMask(play); maskIdx--;
// clang-format on
```

Comment (Dragorn421) [reply src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.c:188]: looks more plausible but formatting oddities suck so not sure if it's overall better
I guess it is slightly?

Comment (fig02) [reply src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.c:188]: either one can have a fake match comment for now and see if anyone can find something better during a fake match pass

Comment (cadmic) [reply src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.c:189]: Alright, I left it as is and left the alternative in a comment

**Review by Dragorn421 — APPROVED** [review]

**Pull request merged**